File: 백준/Silver/10866. 덱/덱.py
import sys
from collections import deque
input = sys.stdin.readline
t = int(input())
q = deque()
for _ in range(t):
  cmd, *val = input().split()
  if cmd == 'push_front':
    num = int(val[0])
    q.appendleft(num)
  elif cmd == 'push_back':
    num = int(val[0])
    q.append(num)
  elif cmd == 'pop_front':
    print(q.popleft()) if q else print(-1)
  elif cmd == 'pop_back':
    print(q.pop()) if q else print(-1)
  elif cmd == 'empty':
    print(1) if len(q)==0 else print(0)
  elif cmd == 'size':
    print(len(q))
  elif cmd == 'front':
    print(q[0]) if q else print(-1)
  elif cmd == 'back':
    print(q[-1]) if q else print(-1)
  

File: 백준/Silver/10866. 덱/덱.js
class Deque {
  constructor() {
    this.deque = [];
  }
  push_front(newElement) {
    this.deque.unshift(newElement);
  }
  push_back(newElement) {
    this.deque.push(newElement);
  }

  pop_front() {
    if (this.empty()) return -1;
    return this.deque.shift();
  }
  pop_back() {
    if (this.empty()) return -1;
    return this.deque.pop();
  }
  size() {
    return this.deque.length;
  }
  empty() {
    return this.size() === 0 ? 1 : 0;
  }
  front() {
    return this.empty() ? -1 : this.deque[0];
  }
  back() {
    return this.empty() ? -1 : this.deque.at(-1);
  }
}

const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let [n, ...input] = fs.readFileSync(filePath).toString().trim().split("\n");
const N = +n;
const commands = input.map((row) => row.split(" "));
const deque = new Deque();
const answer = [];

for (let command of commands) {
  // push
  if (command.length === 2) {
    const [c, n] = command;
    deque[c](n);
  } else {
    answer.push(deque[command]());
  }
}

console.log(answer.join("\n"));
